frontend/tests/playwright/msw/request-handler.ts

```diff
@@ -0,0 +1,233 @@
+ * to configure what the response should return if the request is matched.
+ *
+ * @property `status` - The HTTP status code to return
+ * @property `headers` - The headers to return with the response
+ * @property `body` - The body of the response.
+ * @property `responseType` - This the kind of transformation to apply to the
+ * body when it will be returned. This is only applicable to REST requests
+ * since GraphQL responses will only use a `data` transformer.
+ *
+ * @see https://mswjs.io/docs/basics/response-transformer#standards-transformers
+ * @see https://mswjs.io/docs/api/context
+ * @see https://mswjs.io/docs/api/response
+ *
+ */
+export type MockedResponseInfo = {
+   status: number;
+   headers?: Record<string, string | string[]>;
+   body?: Record<string, any> | string | number | boolean | null;
+   responseType?: 'json' | 'text' | 'xml' | 'raw';
+};
+
+/**
+ * These are some options that can be passed to configure the request handler.
+ * @property `once` - If true, the handler will only be used once.
+ * @see https://mswjs.io/docs/api/response/once
+ * @property `passthrough` - If true, the request will be bypassed and performed
+ * as-is.
+ * @see https://mswjs.io/docs/api/request/passthrough
+ * @property `customResolver` - If provided, this will be used as the resolver
+ * for the handler. This is useful if you want to do something more complex than
+ * just returning a static response.
+ * @see https://mswjs.io/docs/basics/response-resolver
+ */
+export type HandlerOptions = {
+   once: boolean;
+   passthrough: boolean;
+   customResolver?: GraphQLResolver | RestResolver;
+};
+
+export default class Handler {
+   /**
+    * Creates a new GraphQL handler.
+    */
+   static create({
+      request,
+      response,
+      options,
+   }: {
+      request: Omit<RequestInfo, 'method'> & { method: GraphQLMethods };
+      response: Omit<
+         MockedResponseInfo,
+         'headers' | 'body' | 'responseType'
+      > & {
+         body: Record<string, any>;
+      };
+      options?: HandlerOptions;
+   }): GraphQLHandler;
+
+   /**
+    * Creates a new Rest Handler.
+    */
+   static create({
+      request,
+      response,
+      options,
+   }: {
+      request: Omit<RequestInfo, 'method'> & { method: RestMethods };
+      response: MockedResponseInfo;
+      options?: HandlerOptions;
+   }): RestHandler;
+
+   static create({
+      request,
+      response,
+      options = { once: false, passthrough: false },
+   }: {
+      request: RequestInfo;
+      response: MockedResponseInfo;
+      options?: HandlerOptions;
+   }): RequestHandler {
+      const { endpoint, method } = request;
+      const { status, headers, body, responseType } = response;
+      const { once, passthrough, customResolver } = options;
+
+      if (isGraphQLMethod(method)) {
+         if (!isRecord(body)) {
+            throw new Error(`Invalid GraphQL body ${body}. Must be an object`);
+         }
```

Comment:
I'm guessing that this check can be made redundant with enough care in typing. I'm not opposed to leaving it even in that case, in order to provide safety against untyped calls, but if TypeScript forced this check, it's probably indicative of a deeper problem.

Reply:
It became redundant in [`cd6fcf4` (#1348)](https://github.com/iFixit/react-commerce/pull/1348/commits/cd6fcf42ebeae5708a4edfa0958a3c48987fe9dd). Prior to that, the `body` type for a GraphQL Handler could have been anything acceptable under a Rest `body` type. That is why the check was there but is no longer needed.